Calculator/Form1.cs

```diff
@@ -190,131 +190,201 @@ private void btn_CE_Click(object sender, EventArgs e)
+            lbl_title.Text = "^";
+        }
 
-        private void btn_result_Click(object sender, EventArgs e)
+        private void btn_SquareRoot_Click(object sender, EventArgs e)
         {
-            lbl_title.Text = "Result";
+            saveFirstTermAndOperation();
+            operationPerformed = CalculatorOperation.SquareRoot;
+            lbl_title.Text = "SqRt";
+        }
+
+        private void btn_NumberParity_Click(object sender, EventArgs e)
+        {
+            saveFirstTermAndOperation();
+            operationPerformed = CalculatorOperation.NumberParity;
+            lbl_title.Text = "NumberParity";
+        }
+
+        private void btn_MirroredNumber_Click(object sender, EventArgs e)
+        {
+            saveFirstTermAndOperation();
+            operationPerformed = CalculatorOperation.MirroredNumber;
+            lbl_title.Text = "MirroredNumber";
+        }
+
+        private void btn_IsPrime_Click(object sender, EventArgs e)
+        {
+            saveFirstTermAndOperation();
+            operationPerformed = CalculatorOperation.IsPrime;
+            lbl_title.Text = "IsPrime";
+        }
+
+        private void btn_PalindromeSuperPalindrome_Click(object sender, EventArgs e)
+        {
+            saveFirstTermAndOperation();
+            operationPerformed = CalculatorOperation.PalindromeSuperPalindrome;
+            lbl_title.Text = "PalindromeSuperPalindrome";
+        }
+        private void btn_LeastCommonMultiple_Click(object sender, EventArgs e)
+        {
+            saveFirstTermAndOperation();
+            operationPerformed = CalculatorOperation.LeastCommonMultiple;
+            lbl_title.Text = "LeastCommonMultiple";
+        }
 
-            secondTerm = ConvertDouble(textbox_result.Text);
+        private void saveFirstTermAndOperation()
+        {
+            isOperationPerformed = true;
+            firstTerm = textbox_result.Text;
+        }
 
+        private void btn_BiggestCommunalDivisor_Click(object sender, EventArgs e)
+        {
+            saveFirstTermAndOperation();
+            operationPerformed = CalculatorOperation.BiggestCommunalDivisor;
+            lbl_title.Text = "BiggestCommunalDivisor";
+        }
+
+        private void btn_result_Click(object sender, EventArgs e)
+        {
+            lbl_title.Text = "Result";
+            secondTerm = textbox_result.Text;
             try
             {
                 PerformOperation();
             }
             catch (CalculatorException ex)
             {
-                HandleCalculatorException(ex);
+                var errorHandler = new CalculatorExceptionHandler();
+                lbl_title.Text = errorHandler.HandleCalculatorException(ex);
             }
         }
 
-        private void HandleCalculatorException(CalculatorException ex)
+        private void PerformOperation()
         {
-            switch (ex.ErrorType)
+            switch (operationPerformed)
```

Comment:
You are doing the same instructions on each case, you don't need the switch anymore, since you do the same thing in each case